server/src/main/java/org/opensearch/extensions/action/ExtensionTransportActionsHandler.java

```diff
@@ -187,7 +195,14 @@ public String executor() {
         } catch (Exception e) {
             logger.info("Failed to send transport action to extension " + extension.getName(), e);
         }
-        inProgressLatch.await(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+        try {
+            inProgressFuture.orTimeout(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof TimeoutException) {
+                logger.info("No response from extension to request.");
+            }
+            throw Exception.class.cast(e.getCause());
```

Comment:
Please replace with the same checks as here https://github.com/opensearch-project/OpenSearch/pull/5646/files#diff-c9a4accc046eb92a0e6a025970b3eacacc330f40bb55b02859df05a79710c8c8R558